### ISSUE TITLE
Duo two-factor needs option for entering a code

### ISSUE THREAD

**Issue by ljl-geek** [hashicorp/vault#6833]
When doing the Vault login, there is only one two-factor method available: Duo Push to the default device.  This is not ideal, to say the least. 

Heavy Duo/two-factor users often use the code generation app or, more commonly, a Yubikey.

Please add an option to enter a code (or select which device gets a push.)  

The code option is more crucial.  It allows people who are in wireless dead zones or restricted "no-cell" areas to still authenticate with a code generated by the Duo app (dead zone) or a Yubikey (no cell service or wireless device required).

The current method is also misleading:

```Please login VAULT as vault user xxxxxxx with DUO device ready:```

Seeing this, I made sure that my Yubikey was plugged in and ready.  Imagine my irritation when my phone beeped for a Duo Push, which meant I had to fish it out, unlock it, and bring up the push app. (Of the three methods (Duo Push, app code or Yubikey), Duo Push is my least preferred. I'm not alone in this.)

I would rather see a standard command line duo dialog:

```Authenticated with partial success.
Duo two-factor login for xxxxxxx

Enter a passcode or select one of the following options:

 1. Duo Push to XXX-XXX-1234
 2. Duo Push to Tablet (Android)

Passcode or option (1-2): xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.....
Success. Logging you in...```

(The longer passcode is from a Yubikey.)

Thank you for your attention to this.

**Comment by hawleyOSU** [comment]
This is important for Duo users, to maintain the login-flows they are accustomed to. Especially for users who use YubikeyOTP. The typical login flow Duo users expect is to be able to select between multiple MFA authentication at login. The expected behavior is different for text based CLI/API a web based (UI) authentications. 

For CLI/API Authentications a common method is to prompt the user for input, if the user enters an OTP the OTP is is sent to DUO, if the user enters the string "push" a push is sent to the user's primary device. 

For Web logins the user is typically presented a list of available devices to select from. This my be difficult since underlying Vault code would be the same. A reasonable compromise would be to utilize the "Passcode reminder" box for UI logons. The displayed input box should accept an OTP (mobile, hard Token, SMS, YubikeyOTP, or Bypass code) or the string "push". Admins could opt for the current auto-push behavior or enable the "passcode reminder" for more authentication options.

**Comment by hsimon-hashicorp** [comment]
Hello! As of 1.10.0, [Login MFA ](https://developer.hashicorp.com/vault/docs/auth/login-mfa)and integration with Duo was improved. Are you still having issues with using passcodes? Please let me know. Thanks!

**Comment by hawleyOSU** [comment]
Yes, as of 1.13 (and likely 1.15) users can not choose between push or passcode, and all pushes go to the first Duo device. If the passcode reminder is enabled a UI login fails.
